### PR TITLE
Use organization name for SP in disclaimer text

### PR DESCRIPTION
**Pull request by MKodde**
The following fallback sequence was implemented.

Show:
1. Organization Display Name
2. Organization Name when 1 is not available
3. 'Unknown' when 1 and 2 are not available

### CODE REVIEW

File: src/OpenConext/EngineBlock/Metadata/Entity/ServiceProvider.php [hunk 1]
```diff
@@ -364,6 +364,23 @@ public function getDisplayName($preferredLocale = '')
...
+        } elseif ($preferredLocale === 'en') {
+            $orgName = $this->organizationEn->displayName;
+        } elseif ($preferredLocale === 'pt') {
+            $orgName = $this->organizationPt->displayName;
+        }
+
+        if (empty($orgName)) {
+            $orgName = $this->getDisplayName($preferredLocale);
+        }
+        return $orgName;
```
Comment (Badlapje) [inline src/OpenConext/EngineBlock/Metadata/Entity/ServiceProvider.php:381]: is the "unknown" fallback not missing here?

File: src/OpenConext/EngineBlock/Metadata/Entity/ServiceProvider.php [hunk 2]
```diff
@@ -364,6 +364,23 @@ public function getDisplayName($preferredLocale = '')
         return $spName;
     }
 
+    public function getOrganizationName(string $preferredLocale = ''): string
+    {
+        $orgName = '';
+        if ($preferredLocale === 'nl') {
```
Comment (Badlapje) [inline src/OpenConext/EngineBlock/Metadata/Entity/ServiceProvider.php:370]: You could get rid of the first if by making dutch the default locale and overwriting orgName if the preferred locale is en or pt?

I also think using a switch would be better for extendability / readability.  Even better would be to use dynamic method names with something like this (going from memory here, not sure if the () after $locale on the second line is necessary)

```
$locale = "organization${ucFirst($preferredLocale)}";
$orgName = $this->$locale()->displayName;
```

**Review by Badlapje — CHANGES_REQUESTED** [review]
I seem to be missing the "unknown" fallback?

File: src/OpenConext/EngineBlock/Metadata/Entity/ServiceProvider.php (hunk 1 above)
Comment (MKodde) [reply src/OpenConext/EngineBlock/Metadata/Entity/ServiceProvider.php:381]: I'm catching that in the template. That way we can prevent translations ending up in the code. Alternatively I could return the translation key here. But think this is the most user friendly solution for now.

File: src/OpenConext/EngineBlock/Metadata/Entity/ServiceProvider.php (hunk 2 above)
Comment (MKodde) [reply src/OpenConext/EngineBlock/Metadata/Entity/ServiceProvider.php:370]: I'm aware of the possibilities PHP provides in making this method more condensed. And I'm tempted to rewrite the method. 

However, for now I'd like to keep this method as it is, given it is very similar to the 'getDisplayName' method. Who is structured similarly.

Comment (Badlapje) [reply src/OpenConext/EngineBlock/Metadata/Entity/ServiceProvider.php:370]: sounds good 👍

File: src/OpenConext/EngineBlock/Metadata/Entity/ServiceProvider.php (hunk 1 above)
Comment (Badlapje) [reply src/OpenConext/EngineBlock/Metadata/Entity/ServiceProvider.php:390]: 💯

**Review by Badlapje — APPROVED** [review]

**Pull request merged**